include/pybind11/detail/class.h

```diff
@@ -388,15 +388,19 @@ inline void clear_patients(PyObject *self) {
     auto *instance = reinterpret_cast<detail::instance *>(self);
     auto &internals = get_internals();
     auto pos = internals.patients.find(self);
-    assert(pos != internals.patients.end());
-    // Clearing the patients can cause more Python code to run, which
-    // can invalidate the iterator. Extract the vector of patients
-    // from the unordered_map first.
-    auto patients = std::move(pos->second);
-    internals.patients.erase(pos);
-    instance->has_patients = false;
-    for (PyObject *&patient : patients) {
-        Py_CLEAR(patient);
+
+    if (pos != internals.patients.end()) {
+        // Clearing the patients can cause more Python code to run, which
+        // can invalidate the iterator. Extract the vector of patients
+        // from the unordered_map first.
+        auto patients = std::move(pos->second);
+        internals.patients.erase(pos);
+        instance->has_patients = false;
+        for (PyObject *&patient : patients) {
+            Py_CLEAR(patient);
+        }
+    } else {
+        assert(pos != internals.patients.end());
```

Comment:
How could this assert ever fail given the above conditional?

Reply:
It can't fail and could instead just be by itself instead of in the `else`. The reason I did it this way was to avoid the same check when debug mode is on.